tests/test_modeling_tf_common.py

```diff
@@ -1540,18 +1523,16 @@ def test_keras_fit(self):
         config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
         for model_class in self.all_model_classes:
             model = model_class(config)
-            if not getattr(model, "hf_compute_loss", False) and not _return_type_has_loss(model):
-                continue
             # Test that model correctly compute the loss with kwargs
             prepared_for_class = self._prepare_for_class(inputs_dict.copy(), model_class, return_labels=True)
-            # Is there a better way to remove these decoder inputs?
             # We also remove "return_loss" as this is covered by the train_step when using fit()
             prepared_for_class = {
                 key: val
                 for key, val in prepared_for_class.items()
-                if key
-                not in ("head_mask", "decoder_head_mask", "cross_attn_head_mask", "decoder_input_ids", "return_loss")
+                if key not in ("head_mask", "decoder_head_mask", "cross_attn_head_mask", "return_loss")
             }
+            if "labels" in prepared_for_class and "decoder_input_ids" in prepared_for_class:
+                del prepared_for_class["decoder_input_ids"]
```

Comment:
Just for my understanding - why do we now only remove `decoder_input_ids` if `labels` is present?

Reply:
Long story! That test previously had some weird return type logic to figure out which model classes actually returned losses and were trainable. Making this change broke that logic, but that logic was bad (and my fault) to begin with, so I refactored it. As a result, the test ran in some cases where it wasn't running before, which surfaced the bug that in some cases we were passing both `decoder_input_ids` and `labels` to conditional generation models that don't like that, and often have assertions that throw errors if you pass both. In general, for a basic test to check if `fit()` is working, never passing both of those at the same time seems to be the right choice for our models.